lua/cmake-tools/config.lua

```diff
@@ -1,10 +1,12 @@
 local Path = require("plenary.path")
 local scandir = require("plenary.scandir")
 local Result = require("cmake-tools.result")
-local utils = require("cmake-tools.utils")
+-- local utils = require("cmake-tools.utils") -- Fails lua check. Uncomment this for testing
 local Types = require("cmake-tools.types")
 local variants = require("cmake-tools.variants")
 
+local vim = vim -- Localize LSP errors to a single line: [Undefined global 'vim']
```

Comment:
maybe we don't need this, you can change your `sumneko` settings to:
```lua
return {
  settings = {
    ...
    Lua = {
      diagnostics = {
        globals = { "vim" },
        neededFileStatus = {
          ["codestyle-check"] = "Any",
        },
      },
    },
  },
}
```

Reply:
Ah yes! the global vim.

Reply:
This error is still persistant. I think I am using lua_ls, not sumeko is there another option to disable this?

Reply:
I don't know how to configure with lua_ls, maybe there is an option for it.